### PR TITLE
mpi/attr: refactor mpi functions to separate MPIR code

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

The goal is to separate the actual code from the MPI-layer boiler-plates. The latter can be generated by scripts. Reference #4846.

Because we are going to shift large amount of code, we do this one sub-section at a time. This PR refactors the `src/mpi/attr` sub folder.

[skip warnings]
## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/include/mpir_attr_generic.h [hunk 1]
```diff
@@ -125,9 +125,6 @@ MPICH_API_PUBLIC int MPII_Comm_get_attr(MPI_Comm, int, void *, int *, MPIR_Attr_
 MPICH_API_PUBLIC int MPII_Type_get_attr(MPI_Datatype, int, void *, int *, MPIR_Attr_type);
 MPICH_API_PUBLIC int MPII_Win_get_attr(MPI_Win, int, void *, int *, MPIR_Attr_type);
 
-MPICH_API_PUBLIC int MPII_Comm_get_attr_fort(MPI_Comm, int, void *, int *, MPIR_Attr_type);
```
Comment (raffenet) [inline src/include/mpir_attr_generic.h:128]: This should have the corresponding `.c` file change included in this commit.

**Review by raffenet — CHANGES_REQUESTED** [review]
Commit message for `mpi/attr: normalize mpi functions` contains an outdated statement, since that change now appears to be a standalone commit.
```
Call PMPI_Comm_ functions for deprecated functions: MPI_Attr_get,
MPI_Attr_put, MPI_Attr_delete, MPI_Keyval_create, MPI_Keyval_free.
```

**Comment by hzhou** [comment]
test:mpich/ch4/most
test:mpich/ch3/most

**Comment by hzhou** [comment]
@raffenet I have addressed your comments. I also re-arranged the commits and cleaned them up a little. Can you give it another review?

File: src/include/mpir_attr_generic.h (hunk 1 above)
Comment (hzhou) [reply src/include/mpir_attr_generic.h:128]: fixed.

**Review by raffenet — APPROVED** [review]
LGTM

**Pull request merged**